natsstreaming/nats_streaming.go

```diff
@@ -24,24 +24,54 @@ const (
 	OffsetNewest int64 = -1
 )
 
+// ConnectionTimeOutConfig encapsulates interval and timeout seconds fields
+// used for configuring nats streaming connection loss detection
+type ConnectionTimeOutConfig struct {
+	// Number of seconds to wait for a response from nats streaming
+	Seconds int
+	// Number of attempts to establish a connection
+	Tries int
+}
+
 // AsyncMessageSinkConfig is the configuration parameters for an
 // AsyncMessageSink.
 type AsyncMessageSinkConfig struct {
 	URL       string
 	ClusterID string
 	ClientID  string
 	Subject   string
+	TimeOut   ConnectionTimeOutConfig
 }
 
 func NewAsyncMessageSink(config AsyncMessageSinkConfig) (substrate.AsyncMessageSink, error) {
-	sink := asyncMessageSink{subject: config.Subject}
+	sink := asyncMessageSink{subject: config.Subject, connectionLost: make(chan error, 1)}
 
 	clientID := config.ClientID
 	if clientID == "" {
 		clientID = generateID()
 	}
+	var (
+		timeOutSeconds int
+		timeOutTries   int
+	)
+	if config.TimeOut.Seconds < 1 {
+		timeOutSeconds = 1
+	} else {
+		timeOutSeconds = config.TimeOut.Seconds
+	}
+	if config.TimeOut.Tries < 3 {
+		timeOutTries = 3
```

Comment:
I missed this earlier.  Why can't we have tries < 3?

Reply:
not supported

Reply:
https://github.com/nats-io/go-nats-streaming/blob/master/stan.go#L245